src/Raven.Studio/typescript/viewmodels/database/documents/editDocumentConnectedDocuments.ts

```diff
@@ -208,6 +250,53 @@ class connectedDocuments {
         return deferred.promise();
     }
 
+    fetchCounters(skip: number, take: number): JQueryPromise<pagedResult<counterItem>> {
+        const doc = this.document();
+
+        if (!doc.__metadata.hasFlag("HasCounters")) {
+            return connectedDocuments.emptyDocResult<counterItem>();
+        }
+
+        const fetchTask = $.Deferred<pagedResult<counterItem>>();
+        new getCountersCommand(doc.getId(), this.db())
+            .execute()
+            .done(result => {
+                const mappedResults = result.Counters.map(x => this.resultItemToCounterItem(x));               
+              
+                fetchTask.resolve({
+                    items: mappedResults,
+                    totalResultCount: result.Counters.length
+                });
+               
+            })
+            .fail(xhr => fetchTask.reject(xhr));
+
+        return fetchTask.promise();
+    }
+
+    private resultItemToCounterItem(counterDetail: Raven.Client.Documents.Operations.Counters.CounterDetail): counterItem {
```

Comment:
looks like it should to moved to counterItem as constuctor or factory method.

Reply:
counterItem is Interface